lib/ecto/tenant.ex

```diff
@@ -93,39 +60,26 @@ defmodule Ecto.Tenant do
             Process.get(:"$callers", [])
             |> Enum.find_value(fn pid ->
               {:dictionary, dictionary} = Process.info(pid, :dictionary)
-              case Keyword.get(dictionary, @current_tenant_key, :undefined) do
-                :undefined -> false
-                tenant -> tenant
-              end
+              Enum.find_value(dictionary, fn
+                {@current_tenant_key, name} -> name
+                _ -> false
+              end)
             end)
+
           tenant -> tenant
         end
       end
 
       def default_options(_) do
-        tenant = get_tenant() |> tenant_config()
-
-        case tenant do
-          nil -> []
-          tenant -> [prefix: tenant[:prefix]]
-        end
+        config = get_tenant() |> tenant_config()
+        [prefix: config[:prefix]]
       end
 
       defoverridable [get_dynamic_repo: 0]
-      def get_dynamic_repo() do
-        case Process.get({__MODULE__, :dynamic_repo}, :undefined) do
-          :undefined ->
-            Process.get(:"$callers", [])
-            |> Enum.find_value(fn pid ->
-              {:dictionary, dictionary} = Process.info(pid, :dictionary)
 
-              case Enum.find(dictionary, fn {key, _} -> key == {__MODULE__, :dynamic_repo} end) do
-                nil -> false
-                {_, repo} -> repo
-              end
-            end)
-          repo -> repo
-        end
+      def get_dynamic_repo() do
+        config = get_tenant() |> tenant_config()
+        config[:dynamic_repo] || __MODULE__
```

Comment:
We don't need to iterate through the process callers again here because the dynamic repo is part of the tenant config?

Reply:
Yup.